types/time/duration.go

```diff
@@ -0,0 +1,41 @@
+package time
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+// DurationPretty is a wrapper around time.Duration implementing custom
+// marshaller/unmarshaller which make it pretty (e.g. "10s", not
+// "10000000000").
+type DurationPretty struct {
```

Comment:
Is there some reason this has to be a struct? Doesn't that change how this field would be amino encoded compared to using `type DurationPretty time.Duration`? 

Ideally the Amino encoding of structs in `types` matches the protobuf encoding of the types in `abci/types`

Reply:
Hmm. I don't necessarily think its bad to have the custom marshalling so it reads nicer, but just want the Amino marshalling to be consistent. So either we'd need to implement the MarshalBinary methods (does Amino even let you do that yet?) or make it not wrapped in a struct.

We could remove the prettifying all-together because it is JSON after all, which is less intended for human modification. But seeing "100ms" will always be nicer than "100000000" ... 

I think ideally even it would just be in milliseconds, so eg. "100" would be fine, but not sure we can control that without wrapping the type anyways!

Reply:
Amino gives you the possibility of custom marhsalling via implementing `MarshalAmino` / `UnmarshalAmino` (and `MarshalJSON` / `UnmarshalJSON`)

Reply:
See a simple example here: https://gist.github.com/Liamsi/01729f51259f912b0dd64e686025eaee

Reply:
> Ideally the Amino encoding of structs in types matches the protobuf encoding of the types in abci/types

I believe the way protobuf encodes google.protobug.Duration is different from how amino encodes time.Duration.

Reply:
Looks like the simplest thing to do is to make TimeIota int64 ms.